pkg/validation/internal/community.go

```diff
@@ -0,0 +1,277 @@
+package internal
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/blang/semver"
+	"io/ioutil"
+	"os"
+	"strings"
+
+	"github.com/operator-framework/api/pkg/manifests"
+	"github.com/operator-framework/api/pkg/validation/errors"
+	interfaces "github.com/operator-framework/api/pkg/validation/interfaces"
+)
+
+// IndexImagePathKey defines the key which can be used by its consumers
+// to inform where their index image path is to be checked
+const IndexImagePathKey = "index-path"
+
+// OCP version where the apis v1beta1 is no longer supported
+const ocpVerV1beta1Unsupported = "4.9"
+
+const ocpLabelindex = "com.redhat.openshift.versions"
+
+// CommunityOperatorValidator validates the bundle manifests against the required criteria to publish
+// the projects on the community operators
+//
+// Note that this validator allows to receive a List of optional values as key=values. Currently, only the
+// `ocp-version` key is allowed. If informed, it will perform the checks against this specific Kubernetes version where the
+// operator bundle is intend to be distribute for.
+var CommunityOperatorValidator interfaces.Validator = interfaces.ValidatorFunc(communityValidator)
+
+func communityValidator(objs ...interface{}) (results []errors.ManifestResult) {
+
+	// Obtain the k8s version if informed via the objects an optional
+	var indexImagePath = ""
+	for _, obj := range objs {
+		switch obj.(type) {
+		case map[string]string:
+			indexImagePath = obj.(map[string]string)[IndexImagePathKey]
+			if len(indexImagePath) > 0 {
+				break
+			}
+		}
+	}
+
+	for _, obj := range objs {
+		switch v := obj.(type) {
+		case *manifests.Bundle:
+			results = append(results, validateCommunityBundle(v, indexImagePath))
+		}
+	}
+
+	return results
+}
+
+// validateCommunityBundle will check the bundle against the community-operator criterias
+func validateCommunityBundle(bundle *manifests.Bundle, indexImagePath string) errors.ManifestResult {
+	result := errors.ManifestResult{Name: bundle.Name}
+	if bundle == nil {
+		result.Add(errors.ErrInvalidBundle("Bundle is nil", nil))
+		return result
+	}
+
+	if bundle.CSV == nil {
+		result.Add(errors.ErrInvalidBundle("Bundle csv is nil", bundle.Name))
+		return result
+	}
+
+	// Check if has deprecated apis then, check the olm.maxOpenShiftVersion property
+	if len(bundle.V1beta1CRDs) > 0 {
```

Comment:
The list of 1.22 deprecations is larger than just v1beta1crds:

https://kubernetes.io/docs/reference/using-api/deprecation-guide/#v1-22

We should check the full set of resources that were allowed by the Bundle spec rather than just this one API:

customresourcedefinition
clusterrole
role
priorityclass

See here for the list of resource types that are allowed currently:
https://github.com/operator-framework/api/blob/master/pkg/validation/internal/object.go#L16

Reply:
Hi @kevinrizza, 

That is great that you raise it out. We really need to check them as well.